Move some variable ops out of VariableTypeManual.

ailzhang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54543 Move requires_grad_/backward out of VariableTypeManual.
* **#54459 Move some variable ops out of VariableTypeManual.**

Differential Revision: [D27321820](https://our.internmc.facebook.com/intern/diff/D27321820)

Commit:
[TEST] Move some variable ops off dispatcher.

[ghstack-poisoned]

Commit:
Update on "[TEST] Move some variable ops off dispatcher."




[ghstack-poisoned]

Commit:
Update on "Move some variable ops out of VariableTypeManual."




[ghstack-poisoned]

ezyang commented on torch/csrc/jit/api/module.cpp:
```diff
@@ -78,7 +78,7 @@ void Module::to(at::Device device, bool non_blocking) {
 
 void module_state_to(
-    const autograd::Variable& variable,
+    autograd::Variable& variable,
```
What's going on here?

albanD commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -50,16 +50,16 @@
 # is what the previous interpretation was.)
 - func: set_data(Tensor(a!) self, Tensor new_data) -> ()
-  manual_kernel_registration: True
+  manual_cpp_binding: True
```
Why can't we use regular cpp bindings here?

albanD commented on aten/src/ATen/templates/TensorBody.h:
```diff
@@ -790,6 +790,19 @@ class TORCH_API Tensor {
+
+  int64_t output_nr() const;
+
+
```
nit: extra empty line?

Commit:
Update on "Move some variable ops out of VariableTypeManual."




[ghstack-poisoned]

Commit:
Update on "Move some variable ops out of VariableTypeManual."


Differential Revision: [D27321820](https://our.internmc.facebook.com/intern/diff/D27321820)

[ghstack-poisoned]

Commit:
Update on "Move some variable ops out of VariableTypeManual."


Differential Revision: [D27321820](https://our.internmc.facebook.com/intern/diff/D27321820)

[ghstack-poisoned]

Commit:
Update on "Move some variable ops out of VariableTypeManual."


Differential Revision: [D27321820](https://our.internmc.facebook.com/intern/diff/D27321820)

[ghstack-poisoned]
